codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsNaptrRecord.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.internal.StringUtil;
+
+import java.util.regex.Pattern;
+
+/**
+ * Created by epenorr on 2016-08-06.
+ */
+public class DefaultDnsNaptrRecord extends AbstractDnsRecord {
+
+    private final int order;
+
+    private final int preference;
+
+    private final String flags;
+
+    private final String services;
+
+    private final Pattern regexp;
+
+    private final String replacement;
+
+    public DefaultDnsNaptrRecord(String name, DnsRecordType type, int dnsClass, long timeToLive, ByteBuf in) {
```

Comment:
Not sure I agree, the best place to put the knowledge of the record layout is in the record itself. But I'll use the style already established by AbstractDnsRecord